--- ssl/tls13_record_layer.c.orig	2024-09-02 21:43:31.473823653 +0200
+++ ssl/tls13_record_layer.c	2024-09-02 21:45:21.378117189 +0200
@@ -332,6 +332,13 @@
 	return ret;
 }
 
+void
+tls13_record_layer_alert_sent_callback(struct tls13_record_layer *rl,
+    uint8_t alert_level, uint8_t alert_desc)
+{
+       rl->cb.alert_sent(alert_level, alert_desc, rl->cb_arg);
+}
+
 static ssize_t
 tls13_record_layer_send_alert(struct tls13_record_layer *rl)
 {
@@ -361,7 +368,8 @@
 		ret = TLS13_IO_ALERT;
 	}
 
-	rl->cb.alert_sent(rl->alert_level, rl->alert_desc, rl->cb_arg);
+       tls13_record_layer_alert_sent_callback(rl, rl->alert_level,
+           rl->alert_desc);
 
 	return ret;
 }
